kimchi/src/tests/framework.rs

```diff
@@ -124,40 +122,51 @@ impl TestFramework {
     }
 }
 
-impl TestRunner {
+impl<G: KimchiCurve> TestRunner<G>
+where
+    G::ScalarField: PrimeField + Clone,
+    G::BaseField: PrimeField + Clone,
+{
     #[must_use]
-    pub(crate) fn runtime_tables(mut self, runtime_tables: Vec<RuntimeTable<Fp>>) -> Self {
+    pub(crate) fn runtime_tables(
+        mut self,
+        runtime_tables: Vec<RuntimeTable<G::ScalarField>>,
+    ) -> Self {
         self.0.runtime_tables = runtime_tables;
         self
     }
 
     #[must_use]
-    pub(crate) fn recursion(mut self, recursion: Vec<RecursionChallenge<Vesta>>) -> Self {
+    pub(crate) fn recursion(mut self, recursion: Vec<RecursionChallenge<G>>) -> Self {
         self.0.recursion = recursion;
         self
     }
 
-    pub(crate) fn prover_index(&self) -> &ProverIndex<Vesta> {
+    pub(crate) fn prover_index(&self) -> &ProverIndex<G> {
         self.0.prover_index.as_ref().unwrap()
     }
 
     /// Create and verify a proof
-    pub(crate) fn prove_and_verify(self) {
+    pub(crate) fn prove_and_verify<EFqSponge, EFrSponge>(self)
+    where
+        EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
```

Comment:
I wish I didn't know how painful this was to get working :smiling_face_with_tear:

Reply:
hahaha

Reply:
Some memories can't be forgotten.. ;-)